docs/advanced/persistence-and-collaboration.mdx

```diff
@@ -0,0 +1,102 @@
+from dara.components import Input
+
+# Tag the variable to be 'collaborative' and have a shared server state
+collab_variable = Variable(default=1, store=BackendStore())
+
+# Input value will automatically be kept in sync 'live' for all users
+Input(value=collab_variable)
+```
+
+Under the hood the `BackendStore` mechanism is very simple:
+- the store data is initialized with the `default` `Variable` value
+- on initial page load, the `Variable` value is fetched from the server
+- whenever a client makes an update, it makes a request to the server to update the state,
+  which then notifies all other clients about the new state
+
+You can interact with the store programmatically to read, write or delete data from the store.
+
+```python
+from dara.core import Variable
+from dara.core.persistence import BackendStore
+from dara.components import Input
+
+store = BackendStore()
+collab_variable = Variable(default='default value', store=store)
+
+# get current value in the store
+value = await store.read()
+# store can also be accessed from the variable
+value = await variable.store.read()
+
+# write a new value to the store
+await store.write('new value')
+
+# delete the existing value from the store
+await store.delete()
+
+# get all value from the store - see `scope` section below for details
+await store.get_all()
+```
+
+The behaviour of the store can be customized by passing arguments to the `BackendStore`.
+
+### `scope`
+
+Scope controls whether the data in a `BackendStore` is shared among all users (`scope='global'`) or separate and specific to a
+given user (`scope='user'`).
+
+When using `global` scope, the state is shared and synced among the users so a change being made to an attached `Variable`
+in one client will immediately be propagated to other users. In this mode, the `get_all` API method will return a dictionary
+of the shape `{'global': 'value'}`.
+
+When using `user` scope, each user has their own copy of the date. A change made to an attached `Variable` in one client will
+only be propagated to other sessions open **for that particular user**, e.g. among all currently open tabs for that user.
+In this mode, the `get_all` API returns a dictionary with user values keyed by the user identifiers, e.g. `{'user1': 'value1', 'user2': 'value2'}`.
+
+:::note
+
+In `user` scope, the API methods `read`, `write` and `delete` can only be used in an authenticated context, i.e. within
+`py_component`s, `DerivedVariable` resolvers or `action`s. This is because these APIs will look up the currently operating
+user and act on their behalf, e.g. `write` would update the value stored only for that particular user.
+
+:::
+
+### `backend`
+
+You can select a particular backend implemention to be used for storage. By default `BackendStore` uses the
+`dara.core.persistence.InMemoryBackend` implementation which simply stores the data in-memory.
+
+Alternatively, you can use the `FileBackend` which uses a JSON file for storage. This enables persisting
```

Comment:
Could we have a little example using the FileBackend? I think initially that will be a common one to use

Reply:
Good point, will add an example